archive/archive.go

```diff
@@ -0,0 +1,146 @@
+package archive
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+)
+
+type fileCreator interface {
+	create(filename string) (*os.File, error)
+}
+type fileOpener interface {
+	open(filename string) (*os.File, error)
+}
+type fileCopier interface {
+	copy(dst io.Writer, src io.Reader) (written int64, err error)
+}
+
+type zipWriter interface {
+	Create(name string) (io.Writer, error)
+	Close() error
+}
+
+type defaultFileHandler struct{}
+
+func (d *defaultFileHandler) open(filename string) (*os.File, error) {
+	return os.Open(filename)
+}
+
+func (d *defaultFileHandler) create(filename string) (*os.File, error) {
+	return os.Create(filename)
+}
+
+func (d *defaultFileHandler) copy(dst io.Writer, src io.Reader) (written int64, err error) {
+	return io.Copy(dst, src)
+}
+
+type ArchiveHandler interface {
+	Open(path string) (io.Writer, error)
+	AppendFileToArchive(path string, writer zip.Writer) error
+	Close() error
+}
+
+// SupportArchiveHandler
+// The normal procedure should look like this.
+// 		1. CreateZipArchiveFile
+// 		2. InitialiseZipWriter
+// 		3. AppendFileToArchive (n times)
+// 		4. Close
+type SupportArchiveHandler struct {
+	writer      zipWriter
+	fileCreator fileCreator
+	fileOpener  fileOpener
+	fileCopier  fileCopier
+}
+
+func New() *SupportArchiveHandler {
+	return &SupportArchiveHandler{
+		fileCreator: &defaultFileHandler{},
+		fileOpener:  &defaultFileHandler{},
+		fileCopier:  &defaultFileHandler{},
+	}
+}
```

Comment:
The function name `New()` is not convincing to me.
New what? 
If the struct you initialize would have the same name as the package, it was ok to me. But the name of the package is `support` and the Struct is named `SupportArchiveHandler`.
So when i read the code i would expect that a new `archive` is created, when the function is called, but actually it is a SupportArchiveHandler.